kafka/consumer.go

```diff
@@ -528,8 +529,15 @@ func (c *Consumer) ReadMessage(timeout time.Duration) (*Message, error) {
 // Close Consumer instance.
 // The object is no longer usable after this call.
 func (c *Consumer) Close() (err error) {
-	if !atomic.CompareAndSwapUint32(&c.isClosed, 0, 1) {
-		return getOperationNotAllowedErrorForClosedClient()
+	// Check if the client is already closed.
+	err = c.verifyClient()
+	if err != nil {
+		return err
+	}
+
+	// Client is in the process of closing.
+	if !atomic.CompareAndSwapUint32(&c.isClosing, 0, 1) {
+		return newErrorFromString(ErrState, "Consumer is already closing")
```

Comment:
Added new error code changes the contract. Can we use old error?

Reply:
The error code is the same (ErrState), but the error message is different. The user is expected to check the error code if doing any error-specific handling.
The string "Operation not allowed on closed client" is not correct for this error, what do you think?

Reply:
If the errorcode is same I think it should be fine.